client/app/src/main/java/uk/ac/cam/cl/echo/extrusionfinder/client/ImageCaptureActivity.java

```diff
@@ -153,4 +169,133 @@ public void onDismiss(DialogInterface dialog) {
 
         dialog.show();
     }
+
+    /**
+     * Callback for when OpenCV has been loaded
+     */
+    private final BaseLoaderCallback openCVLoaderCallback = new BaseLoaderCallback(this) {
+        @Override
+        public void onManagerConnected(int status) {
+            if(status == BaseLoaderCallback.SUCCESS) {
+                // on success flag that we are ready to process frames
+                profileDetectionReady = true;
+            }
+            else {
+                super.onManagerConnected(status);
+            }
+        }
+    };
+
+    /**
+     * Creates and starts an async task to pre-process the image and draw the extrusion highlights
+     * to a canvas.
+     * @param image byte[] NV21 image to pre-process
+     * @param size Dimension representing the size of the image
+     */
+    private void executePreprocessAsyncTask(byte[] image, Dimension size) {
+        AsyncTask<Pair<Pair<byte[], Dimension>, SurfaceHolder>, Void, Void> preprocessAsyncTask
```

Comment:
As we all know, there's no need to write maintainable code when you can be lazy.